service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementMapper.java

```diff
@@ -0,0 +1,202 @@
+        if (referralRequest.hasServiceRequested()) {
+            return SERVICE_REQUESTED + extractServiceRequested(referralRequest);
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String buildRequesterDescription(ReferralRequest referralRequest) {
+        if (referralRequest.hasRequester() && referralRequest.getRequester().getAgent().hasReference()) {
+            IIdType reference = referralRequest.getRequester().getAgent().getReferenceElement();
+            if (reference.getResourceType().equals(ResourceType.Device.name())) {
+                return messageContext.getInputBundleHolder()
+                    .getResource(reference)
+                    .map(resource -> (Device) resource)
+                    .map(value -> extractDevice(value))
+                    .map(text -> REQUESTER_DEVICE + text)
+                    .orElse(StringUtils.EMPTY);
+            } else if (reference.getResourceType().equals(ResourceType.Organization.name())) {
+                return messageContext.getInputBundleHolder()
+                    .getResource(reference)
+                    .map(resource -> (Organization) resource)
+                    .map(value -> REQUESTER_ORG + value.getName())
+                    .orElse(StringUtils.EMPTY);
+            } else if (reference.getResourceType().equals(ResourceType.Patient.name())) {
+                return REQUESTER_PATIENT;
+            } else if (reference.getResourceType().equals(ResourceType.RelatedPerson.name())) {
+                return messageContext.getInputBundleHolder()
+                    .getResource(reference)
+                    .map(resource -> (RelatedPerson) resource)
+                    .map(value -> REQUESTER_RELATION + extractHumanName(value.getNameFirstRep()))
+                    .orElse(StringUtils.EMPTY);
+            }
+        }
+
+        return StringUtils.EMPTY;
+    }
+
+    private String buildSpecialtyDescription(ReferralRequest referralRequest) {
+        Optional<String> specialty = Optional.empty();
+        if (referralRequest.hasSpecialty()) {
+            CodeableConcept specialtyObject = referralRequest.getSpecialty();
+            specialty = extractTextOrCoding(specialtyObject);
+        }
+        return specialty.map(value -> SPECIALTY + value).orElse(StringUtils.EMPTY);
+    }
+
+    private String buildRecipientDescription(ReferralRequest referralRequest) {
+        if (referralRequest.hasRecipient()) {
+            var ignoreFirstPractitioner = removeFirstPractitionerReference(referralRequest.getRecipient());
+
+            return ignoreFirstPractitioner.stream()
+                .filter(Reference::hasReferenceElement)
+                .map(value -> extractRecipient(messageContext, value))
+                .collect(Collectors.joining(COMMA));
+        }
+
+        return StringUtils.EMPTY;
+    }
+
+    private List<Reference> removeFirstPractitionerReference(List<Reference> referenceList) {
```

Comment:
Don't throw it out, we'll need it later! ;) 

How about
```
referralRequest.getRecipient()
  .stream()
  .skip(1)
```
instead?

Reply:
Oh its more complicated than that isn't it. We want to skip the first occurrence of a specific resource.

In that case, create a new list from `referralRequest.getRecipient()` but don't modify the original HAPI FHIR objects. We don't want any data going missing.